agent/config/config.go

```diff
@@ -40,6 +40,20 @@ type ListenerConfig struct {
 	// Timeout is the timeout to forward incoming requests to the upstream.
 	Timeout time.Duration `json:"timeout" yaml:"timeout"`
 
+	// Keep-alive timeout for connections to upstream. Defaults to 30 seconds.
+	KeepAlive time.Duration `json:"keep_alive" yaml:"keep_alive"`
```

Comment:
Can you group these under `http_client`, otherwise the listener config gets quite complex, plus it's HTTP specific?

Such as:
```
listeners:
  - endpoint_id: "123"
    addr: "http://localhost:9208"
    timeout: 7m
    protocol: http
    http_client:
      keep_alive: 10m
```

Reply:
Makes sense -- would you want all the new settings in this PR grouped under the `http_client` property?

Reply:
yep thanks